### PR TITLE
NO-ISSUE: Better documentation for subsystem tests

### PR DESCRIPTION
**Pull request by omertuc**
The service documentation for how to run subsystem tests in kube-api
mode was missing. This commit refactors the subsystem test documentation
to improve clarity & explain how to run them for kube-api mode.

It also changes some variables / target names in the Makefile to give
them more descriptive names

- Improved the `get_service` `Makefile` function to use structured `jq`
parsing rather than grep/awk from raw `kubectl get` human output and
renamed it to `get_service_ip_port`

- Renamed `LOCAL_ASSISTED_ORG` to `SUBSYSTEM_LOCAL_REGISTRY`

- Renamed `LOCAL_SERVICE` to `LOCAL_SERVICE_IMAGE`

- Aliased `deploy-test` target to more descriptive `deploy-service-for-subsystem-test`

- Aliased `test` target to more descriptive `subsystem-test`

- Aliased `test-kube-api` target to more descriptive `subsystem-test-kube-api`

- Removed old `deploy-onprem-for-subsystem` and `test-onprem` `Makefile` targets
    - They seem very old and there doesn't seem to be any obvious way to
      run subsystem on non-kubernetes clusters anyway - `get_service`
      assumes that you're running the service on k8s so `make test` can't
      possibly work against on-prem
    - `test-onprem` seems to be using 5432 as a database port which is
      no longer the port used in the on-prem database, which leads me to
      believe this target is old, unused and unmaintained

    If anyone ever wants to restore subsystem tests on on-prem, feel free
    to revive those targets and update the document accordingly to
    explain how they can be used to run REST-API tests (obviously
    kube-api subsystem won't work on on-prem since there's no k8s)

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [x] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @filanov 
/cc @paul-maidment 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/dev/running-test.md
Comment (filanov) [inline docs/dev/running-test.md:39]: can the sleep be replaced by some wait logic?

File: docs/dev/running-test.md
Comment (filanov) [inline docs/dev/running-test.md:86]: doesn't look related to subsystem flow, it may be confusing no?

File: docs/dev/running-test.md
Comment (omertuc) [reply docs/dev/running-test.md:39]: There is wait logic right underneath it

Comment (filanov) [reply docs/dev/running-test.md:39]: then why do you need this sleep?

Comment (omertuc) [reply docs/dev/running-test.md:39]: Just to reduce the number of iterations (and therefore echos) that the loop below it will do. It's not too important, I'll remove it if it's confusing

File: docs/dev/running-test.md
Comment (omertuc) [reply docs/dev/running-test.md:86]: Right

File: docs/dev/running-test.md
Comment (filanov) [reply docs/dev/running-test.md:39]: or add a comment to why it's there.

File: docs/dev/running-test.md
Comment (omertuc) [reply docs/dev/running-test.md:63]: It is https://github.com/openshift/assisted-service/blob/4c52bddbf10c5312d013ac1a06aa43709361d905/Makefile#L313

**Comment by omertuc** [comment]
Closing to refresh checks

**Comment by omertuc** [comment]
Reopening

File: Makefile
Comment (osherdp) [inline Makefile:412]: we have external usage for the aliases?
because if no, I'd like to just rename and if no then all good (and we should transition with the aliases)

Comment (osherdp) [reply Makefile:412]: or course considering ``deploy-test`` as well

File: docs/dev/running-test.md
Comment (osherdp) [inline docs/dev/running-test.md:42]: seems like this one has specifically tab or something, so indentation seems weird

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by filanov** [comment]
overall LGTM, i'll let @osherdp to approve once his comments are resolved.

File: Makefile
Comment (omertuc) [reply Makefile:412]: >we have external usage for the aliases?

A lot of people may be used to the old names and may be using them in scripts, I don't know. As for `test` and `test-kube-api`, they're both used in `openshift/release`:

```bash
grep "make test" **/*assisted* 2>&1 | grep -v directory

ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:    make test
ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-main.yaml:    make test
ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-release-ocm-2.4.yaml:    make test
ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-release-ocm-2.5.yaml:    make test
ci-operator/config/openshift/assisted-image-service/openshift-assisted-image-service-release-ocm-2.5.yaml:    make test
ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml:        make test
ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:        make test
ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:        make test
ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:        make test-kube-api
ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml:        make test-kube-api
ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.4.yaml:        make test
ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.4.yaml:        make test
ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.4.yaml:        make test-kube-api
ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.4.yaml:        make test-kube-api
ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5.yaml:        make test
ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5.yaml:        make test
ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5.yaml:        make test-kube-api
ci-operator/config/openshift/assisted-service/openshift-assisted-service-release-ocm-2.5.yaml:        make test-kube-api
```

Comment (osherdp) [reply Makefile:412]: Ack
Lots of the references there are for different repositories, but it does demonstrate we have external usage
Thanks!

**Review by osherdp — APPROVED** [review]

**Pull request merged**
